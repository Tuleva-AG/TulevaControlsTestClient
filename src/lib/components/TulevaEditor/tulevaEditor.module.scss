@import "./../../styles/variables.scss";
@import "./../../styles/controls.scss";

.display:hover {
  color: $color-hover;
}

.wrappedButtons {
  white-space: nowrap;
}

.viewTable {
  display: table;
  width: 100%;

  .viewRow {
    display: table-row;
    width: 100%;
  }
}

/* f√ºr Desktop & Tablet */

@media only screen and (min-width: 600px) {
  .viewTable {
    display: table;
    width: 100%;

    .viewRow {
      display: table-row;
      width: 100%;
      div {
        display: table-cell;
        width: 25%;
      }
    }
  }
}
