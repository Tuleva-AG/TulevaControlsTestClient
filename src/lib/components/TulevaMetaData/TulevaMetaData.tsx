import moment from "moment";
import React from "react";
import { IBaseEntity } from "../../types/baseEntity";
import styles from "./tulevaMetaData.module.scss";

export interface ITulevaMetaDataProps {
  item: IBaseEntity;
  labelCreated?: string;
  labelModified?: string;
  labelCreatedby?: string;
  labelModifiedby?: string;
}

const TulevaMetaData: React.FC<ITulevaMetaDataProps> = (
  props: ITulevaMetaDataProps
) => {
  let labelCreated = props.labelCreated ? props.labelCreated : "Erstellt am: ";
  let labelModified = props.labelModified
    ? props.labelModified
    : "Ge√§ndert am: ";
  let labelCreatedby = props.labelCreatedby ? props.labelCreatedby : "von: ";
  let labelModifiedby = props.labelModifiedby ? props.labelModifiedby : "von: ";

  return (
    <div className={styles.labeledControl}>
      <div className={styles.metaDataTable}>
        <div className={styles.metaDataRow}>
          <div className={styles.metaDataLabel}>
            {labelCreated}
            {moment(props.item.created).format("lll")} {props.item.createdBy && labelCreatedby}
            {props.item.createdBy?.name}
          </div>
        </div>
        <div className={styles.metaDataRow}>
          <div>
            {labelModified}
            {moment(props.item.edited).format("lll")} {props.item.editedBy && labelModifiedby}
            {props.item.editedBy?.name}
          </div>
        </div>
      </div>
    </div>
  );
};

export default TulevaMetaData;
