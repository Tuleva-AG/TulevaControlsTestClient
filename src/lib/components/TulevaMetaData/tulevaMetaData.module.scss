.metaDataTable {
  display: table;
  width: 100%;
  height: 40px;
  // border: 1px solid $color-border;
  padding-left: 8px;
  float: left;
  // background-color: ghostwhite;
  .metaDataRow {
    display: table-row;
    div {
      display: table-cell;
      white-space: nowrap;
      font-size: 0.9em;
      font-family: "Segoe UI", "Segoe", Tahoma, Helvetica, Arial, sans-serif;
      text-align: left;
    }
  }
  .metaDataLabel {
    width: 10%;
  }
}
