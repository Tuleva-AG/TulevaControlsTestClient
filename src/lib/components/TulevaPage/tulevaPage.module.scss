@import "./../../styles/variables.scss";

.header {
  display: table;
  width: 100%;
  border: 1px solid $color-border;
  background-color: $color-white;
  padding: 8px 16px;

  margin-bottom: 4px;

  .headerInner {
    display: table-row;
    .headerCaption {
      padding: 8px 16px;
      width: calc(100% - 75px);

      display: table-cell;

      h2 {
        margin: 0;
      }
    }
    .headerButtons {
      width: 75px;
      display: table-cell;
      text-align: right;
      vertical-align: middle;
    }
  }
}

.listItem {
  background-color: $color-white;
  display: table;

  table-layout: fixed;
  width: 100%;
}
.filterOptions {
  margin-bottom: 4px;
}

.viewTable {
  display: grid;
  width: calc(100% - 75px);
  padding: 0px 16px;

  .viewRow {
    display: grid;
    width: 100%;
    div {
      font-weight: bold;
    }
  }
}

@media only screen and (min-width: 600px) {
  .viewTable {
    display: table;
    table-layout: fixed;
    width: calc(100% - 75px);
    padding: 0px 16px;

    .viewRow {
      display: table-row;
      width: 100%;
      div {
        display: table-cell;
        width: 20%;
        font-weight: bold;
        cursor: pointer;
      }
    }
  }
}
