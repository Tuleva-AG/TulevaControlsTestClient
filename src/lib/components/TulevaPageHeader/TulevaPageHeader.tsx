import React, { ReactNode, useState } from "react";
// import HelpTopic from '../../admin/help/helpTopic/helpTopic';
// import { QuestionCircleOutlined } from '@ant-design/icons';

import styles from "./tulevaPageHeader.module.scss";
import { EditorContextProvider } from "../tulevaSharedContext";
// import { useTranslation } from 'react-i18next';

interface IPageHeaderProps {
  caption: string;
  subLine?: string;
  summary?: string;
  children?: ReactNode;
  className?: string;
  helpTopic?: string;
  buttonsClassName?: string;
}

const PageHeader: React.FC<IPageHeaderProps> = (props: IPageHeaderProps) => {
  const [helpVisible, setHelpVisible] = useState<boolean>(false);

  return (
    <div className={styles.header + " " + props.className}>
      <div className={styles.headerInner}>
        <div className={styles.headerCaption}>
          <h2>{props.caption}</h2>
          {/* {props.helpTopic &&
                        <QuestionCircleOutlined title={t('pageHeader.helpTooltip')} onClick={() => setHelpVisible(!helpVisible)} style={{ fontSize: '1.5em', color: '#fffff', display: 'table-cell', paddingLeft: '0.8em', verticalAlign: 'middle' }} />
                    } */}
        </div>
      </div>
      <div className={styles.headerInner}>
        <div>
          <EditorContextProvider>
            {/* {props.helpTopic && helpVisible &&
                            <HelpTopic topic={props.helpTopic as string}></HelpTopic>
                        } */}
          </EditorContextProvider>
        </div>
      </div>
      <div className={styles.headerInner}>
        {
          props.subLine && <div className={styles.headerSubline}>{props.subLine}</div>
        }
        {
          props.children && <div className={styles.headerButtons + " " + props.buttonsClassName}>
            {props.children}
          </div>
        }
        {
          props.summary && <div className={styles.headerSummary}>{props.summary}</div>
        }
      </div>
    </div>
  );
};

export default PageHeader;
