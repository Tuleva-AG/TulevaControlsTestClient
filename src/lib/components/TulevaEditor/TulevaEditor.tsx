import {
  CheckCircleTwoTone,
  CloseCircleTwoTone,
  DeleteOutlined,
  CopyOutlined,
  EditOutlined,
} from "@ant-design/icons";
import { Button, Modal, Popconfirm } from "antd";
import React, { useEffect, useState } from "react";
import { IBaseEntity } from "../../types/baseEntity";
import styles from "./tulevaEditor.module.scss";
import { useEditorContext } from "../tulevaSharedContext";
import TulevaMetaData from "../TulevaMetaData/TulevaMetaData";

export type PartProps = {
  children: React.ReactNode;
};

interface ITulevaEditorProps {
  item: IBaseEntity;
  onSave: (item: IBaseEntity) => void;
  onClick?: (item: IBaseEntity) => void;
  onDelete: (id: string | undefined) => void;
  onCopyItem?: (id: string | undefined) => void;
  editMode: boolean;
  hideEditButton?: boolean;
  hideDeleteButton?: boolean;
  hideCopyButton?: boolean;
  hideExitButton?: boolean;
  hideSaveButton?: boolean;
  hideMetaData?: boolean;
  onExit: (item: IBaseEntity) => void;

  editor?: React.FC<PartProps>;
  onRenderEdit: (item: IBaseEntity) => React.ReactNode;
  onRenderView?: (item: IBaseEntity) => React.ReactNode;

  deleteRights?: boolean;
  editRights?: boolean;

  showAsModal?: boolean;
  stayInEditMode?: boolean;
  confirmOnSave?: boolean;
  confirmOnSaveMessage?: string;
  wrapButtons: boolean;

  onRenderEditorFooter?: (
    item: IBaseEntity,
    closeEditor: (e: any) => void
  ) => React.ReactNode;
  additionalEditButtons?: (item: IBaseEntity) => JSX.Element;
  additionalViewButtons?: (item: IBaseEntity) => JSX.Element;

  editorTitle?: string;
  onRenderEditorCaption?: (item: IBaseEntity) => React.ReactNode;
  labelSave?: string;
  labelCancel?: string;
  labelConfirmDelete?: string;
  labelYes?: string;
  labelNo?: string;
  labelDeleteEntry?: string;
  labelCopy?: string;
  labelEdit?: string;

  customEditButtonContent?: JSX.Element;
}

const TulevaEditor: React.FC<ITulevaEditorProps> = (props) => {
  const getEditorContext = useEditorContext();
  const [editMode, setEditMode] = useState(props.editMode);
  const [itemState, setItemState] = useState(props.item);
  const [confirmState, setConfirmState] = useState<boolean>(false);

  let deletePerms = true;
  if (props.deleteRights != undefined) {
    deletePerms = props.deleteRights;
  }
  let editPerms = true;
  if (props.editRights != undefined) {
    editPerms = props.editRights;
  }

  let labelSave = props.labelSave ? props.labelSave : "Speichern";
  let labelCancel = props.labelCancel ? props.labelCancel : "Abbrechen";
  let labelConfirmDelete = props.labelConfirmDelete
    ? props.labelConfirmDelete
    : "Diesen Eintrag wirklich löschen?";
  let labelYes = props.labelYes ? props.labelYes : "Ja";
  let labelNo = props.labelNo ? props.labelNo : "Nein";
  let labelDeleteEntry = props.labelDeleteEntry
    ? props.labelDeleteEntry
    : "Eintrag löschen...";
  let labelCopy = props.labelCopy ? props.labelCopy : "Kopieren";
  let labelEdit = props.labelEdit ? props.labelEdit : "Bearbeiten";

  const toggleEditMode = (e: any) => {
    if (!props.stayInEditMode) {
      setEditMode(!editMode && editPerms);
    }
    if (itemState.isDirty) {
      props.onSave(itemState);
      setItemState({
        ...itemState,
        isDirty: false,
      });
    }

    e.preventDefault(); // prevent scrolling to top after click
  };

  const confirm = (e: any) => {
    setConfirmState(false);
    try {
      toggleEditMode(e);
    } catch (error) {
      console.log(error);
    }
  };

  const cancel = () => {
    setConfirmState(false);
  };

  const handleVisibleChange = (visible: boolean) => {
    if (props.confirmOnSave) {
      setConfirmState(visible);
    } else {
      toggleEditMode(undefined);
    }
  };

  const exitEditMode = (e: any) => {
    setEditMode(!editMode);
    if (itemState.id?.includes("New_")) {
      props.onDelete(itemState.id);
      return;
    }
    setItemState(props.item);

    if (props.item.id?.includes("New_")) {
      props.onExit(props.item);
    }
    e.preventDefault(); // prevent scrolling to top after click
  };

  useEffect(() => {
    setItemState(getEditorContext.item);
  }, [getEditorContext.item]);

  function renderDefaultViewButtons(): React.ReactNode {
    const editButtonContent = props.customEditButtonContent ? (
      props.customEditButtonContent
    ) : (
      <EditOutlined />
    );

    return (
      <div className={props.wrapButtons ? styles.wrappedButtons : ""}>
        {!props.hideEditButton && (
          <Button
            className={styles.addNew}
            shape="circle"
            size="middle"
            title={labelEdit}
            icon={editButtonContent}
            onClick={toggleEditMode}
          />
        )}
        {!props.hideCopyButton && (
          <Button
            className={styles.addNew}
            shape="circle"
            size="middle"
            title={labelCopy}
            icon={<CopyOutlined />}
            onClick={() => props.onCopyItem && props.onCopyItem(props.item.id)}
          />
        )}
        {!props.hideDeleteButton && (
          <Popconfirm
            title={labelConfirmDelete}
            placement="topRight"
            onConfirm={() => props.onDelete(props.item.id)}
            okText={labelYes}
            cancelText={labelNo}
          >
            <Button
              className={styles.addNew}
              shape="circle"
              size="middle"
              title={labelDeleteEntry}
              icon={<DeleteOutlined />}
              disabled={!deletePerms}
            />
          </Popconfirm>
        )}
        {props.additionalViewButtons && props.additionalViewButtons(props.item)}
      </div>
    );
  }

  return (
    <div className={styles.editorInner}>
      <div className={styles.inputArea}>
        {editMode && itemState && (
          <>
            {props.showAsModal && (
              <>
                {props.onRenderView && (
                  <div
                    className={styles.display + " " + styles.viewTable}
                    onClick={toggleEditMode}
                  >
                    <div className={styles.viewRow}>
                      {props.onRenderView(props.item)}
                    </div>
                  </div>
                )}

                <Modal
                  title={
                    props.onRenderEditorCaption
                      ? props.onRenderEditorCaption(props.item)
                      : props.editorTitle
                  }
                  open={editMode}
                  onOk={toggleEditMode}
                  onCancel={exitEditMode}
                  width={1000}
                  bodyStyle={{ marginBottom: "30px" }}
                  closable={false}
                  maskClosable={false}
                  okButtonProps={{ disabled: !getEditorContext.isValid }}
                  okText={labelSave}
                  cancelText={labelCancel}
                  footer={
                    props.onRenderEditorFooter
                      ? props.onRenderEditorFooter(props.item, toggleEditMode)
                      : undefined
                  }
                >
                  {props.onRenderEdit(props.item)}
                  {!props.hideMetaData && (
                    <TulevaMetaData item={props.item}></TulevaMetaData>
                  )}
                </Modal>
              </>
            )}
            {!props.showAsModal && (
              <>
                {props.onRenderEdit(props.item)}
                {!props.hideMetaData && (
                  <TulevaMetaData item={props.item}></TulevaMetaData>
                )}
              </>
            )}
          </>
        )}
        {itemState && !editMode && props.onRenderView && (
          <div
            className={styles.display + " " + styles.viewTable}
            onClick={(e: any) => {
              props.onClick ? props.onClick(props.item) : toggleEditMode(e);
            }}
          >
            <div className={styles.viewRow}>
              {props.onRenderView(props.item)}
            </div>
          </div>
        )}
      </div>
      <div className={styles.buttonArea}>
        <div className={styles.buttonAreaInner}>
          {props.item && editMode && !props.showAsModal && (
            <>
              {!props.hideSaveButton && (
                <Popconfirm
                  title={props.confirmOnSaveMessage}
                  open={confirmState}
                  placement="topRight"
                  onOpenChange={handleVisibleChange}
                  onConfirm={confirm}
                  onCancel={cancel}
                  okText={labelYes}
                  cancelText={labelNo}
                  disabled={!getEditorContext.isValid}
                >
                  <Button
                    shape="circle"
                    size="middle"
                    title={labelSave}
                    icon={<CheckCircleTwoTone twoToneColor="#52c41a" />}
                    disabled={!getEditorContext.isValid}
                  />
                </Popconfirm>
              )}
              {!props.hideExitButton && (
                <Button
                  shape="circle"
                  size="middle"
                  title={labelCancel}
                  icon={<CloseCircleTwoTone twoToneColor="#f5222d" />}
                  onClick={exitEditMode}
                />
              )}
              {/* only for none modal mode */}
              {props.additionalEditButtons &&
                props.additionalEditButtons(props.item)}
            </>
          )}
          {props.item && editMode && props.showAsModal && (
            <>{renderDefaultViewButtons()}</>
          )}
          {!editMode && renderDefaultViewButtons()}
        </div>
      </div>
    </div>
  );
};

export default TulevaEditor;
