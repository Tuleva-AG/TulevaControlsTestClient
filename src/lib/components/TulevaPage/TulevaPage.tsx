import { PlusCircleOutlined } from "@ant-design/icons";
import { Button, List, Spin } from "antd";
import React, { useEffect, useState } from "react";
import { IBaseEntity } from "../../types";
import TulevaEditor from "../TulevaEditor";
import { PartProps } from "../TulevaEditor/TulevaEditor";
import { EditorContextProvider } from "../tulevaSharedContext";
import PageHeader from "../TulevaPageHeader";
import styles from "./tulevaPage.module.scss";

export interface ITulevaPageProps {
  caption?: string;
  subLine?: string;
  summary?: string;
  helpTopic?: string;
  deleteRights?: boolean;
  editRights?: boolean;
  createRights?: boolean;
  listRights?: boolean;

  hideEditButton?: boolean;
  hideDeleteButton?: boolean;
  hideExitButton?: boolean;
  hideSaveButton?: boolean;
  hideCopyButton?: boolean;
  hideMetaData?: boolean;
  loading: boolean;
  onGetEditMode?: (item: IBaseEntity) => boolean;

  onRenderEdit: (item: IBaseEntity) => JSX.Element;
  onRenderView: (item: IBaseEntity) => JSX.Element;
  onRenderListHeader: () => JSX.Element;
  onRenderFilter: () => JSX.Element;
  onRenderHeaderButtons?: () => JSX.Element;
  dataSource?: IBaseEntity[];

  onClick?: (item: IBaseEntity | any) => void;
  onSave: (item: IBaseEntity | any, id: string) => void;
  onDelete: (id: string | undefined) => void;
  onCopyItem?: (id: string | undefined) => void;

  onAddNew?: () => void;
  showAsModal?: boolean;
  stayInEditMode?: boolean;
  confirmOnSave?: boolean;
  confirmOnSaveMessage?: string;

  onExit: (item: IBaseEntity) => void;

  additionalViewButtons?: (item: IBaseEntity) => JSX.Element;
  additionalEditButtons?: (item: IBaseEntity) => JSX.Element;
  onRenderEditorFooter?: (
    item: IBaseEntity,
    closeEditor: (e: any) => void
  ) => React.ReactNode;

  labelNew?: string;
  editorTitle?: string;
  onRenderEditorCaption?: (item: IBaseEntity) => React.ReactNode;
  wrappedButtons?: boolean;
}

const Header: React.FC<PartProps> = ({ children }) => <>{children}</>;

const Filter: React.FC<PartProps> = ({ children }) => <>{children}</>;

const Editor: React.FC<PartProps> = ({ children }) => <>{children}</>;

const TulevaPage: React.FC<ITulevaPageProps> = (props: ITulevaPageProps) => {
  const [data, setData] = useState<IBaseEntity[]>(
    props.dataSource as IBaseEntity[]
  );
  const createPerms = props.createRights;

  let labelNew = props.labelNew ? props.labelNew : "Neu";

  useEffect(() => {
    setData(props.dataSource as IBaseEntity[]);
  }, [props.dataSource]);

  const onSaveClick = (item: IBaseEntity | any, id: string): void => {
    if (props.onSave !== undefined) {
      props.onSave(item, id);
    }
  };

  const onGetEditMode = (item: IBaseEntity | any): boolean => {
    if (props.onGetEditMode === undefined) {
      return item.id?.includes("New_");
    } else {
      return props.onGetEditMode(item);
    }
  };

  const renderItem = (item: IBaseEntity) => {
    return (
      <List.Item key={item.id} className={styles.listItem}>
        <EditorContextProvider>
          <TulevaEditor
            item={item}
            editMode={onGetEditMode(item)}
            onSave={(saveItem) => onSaveClick(saveItem, item.id as string)}
            onClick={props.onClick}
            onDelete={() => props.onDelete(item.id)}
            onCopyItem={() => props.onCopyItem && props.onCopyItem(item.id)}
            onExit={(): void => props.onExit(item)}
            editor={Editor}
            deleteRights={props.deleteRights}
            editRights={props.editRights}
            hideEditButton={props.hideEditButton}
            hideDeleteButton={props.hideDeleteButton}
            hideExitButton={props.hideExitButton}
            hideSaveButton={props.hideSaveButton}
            hideMetaData={props.hideMetaData}
            hideCopyButton={props.hideCopyButton}
            showAsModal={props.showAsModal}
            stayInEditMode={props.stayInEditMode}
            confirmOnSave={props.confirmOnSave}
            confirmOnSaveMessage={props.confirmOnSaveMessage}
            onRenderView={() => props.onRenderView(item)}
            onRenderEdit={() => props.onRenderEdit(item)}
            additionalEditButtons={props.additionalEditButtons}
            additionalViewButtons={props.additionalViewButtons}
            onRenderEditorFooter={(footerItem, closeEditor) =>
              props.onRenderEditorFooter !== undefined
                ? props.onRenderEditorFooter(footerItem, closeEditor)
                : undefined
            }
            editorTitle={props.editorTitle}
            onRenderEditorCaption={props.onRenderEditorCaption}
            wrapButtons={props.wrappedButtons ? props.wrappedButtons : false}
          ></TulevaEditor>
        </EditorContextProvider>
      </List.Item>
    );
  };

  const onAddNewClick = (): void => {
    if (props.onAddNew !== undefined) {
      props.onAddNew();
    }
  };

  const renderPageHeader = (): JSX.Element => {
    return (
      <>
        <PageHeader
          helpTopic={props.helpTopic}
          caption={props.caption as string}
          subLine={props.subLine}
          summary={props.summary}
        >
          {!props.onRenderHeaderButtons && (
            <Button
              className={styles.addNew}
              shape="circle"
              size="middle"
              title={labelNew}
              icon={<PlusCircleOutlined />}
              onClick={onAddNewClick}
              disabled={!createPerms}
            />
          )}
          {props.onRenderHeaderButtons && props.onRenderHeaderButtons()}
        </PageHeader>
      </>
    );
  };

  const renderListHeader = (): JSX.Element => {
    return (
      <>
        <div className={styles.viewTable}>
          <div className={styles.viewRow}>{props.onRenderListHeader()}</div>
        </div>
        <div className={styles.buttonArea}></div>
      </>
    );
  };

  // if (!useHasPermissions(props.listRights)) {
  //     return <Redirect to={'/'} />;
  // }

  return (
    <div>
      <Header>{renderPageHeader()}</Header>
      <Spin spinning={props.loading} size="large">
        <Filter>{props.onRenderFilter()}</Filter>
        <List
          className={styles.cardStyle}
          size="small"
          bordered
          dataSource={data}
          header={renderListHeader()}
          renderItem={(item) => renderItem(item)}
        />
      </Spin>
    </div>
  );
};

export default TulevaPage;
