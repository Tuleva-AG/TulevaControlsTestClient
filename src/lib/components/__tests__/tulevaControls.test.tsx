import { render } from "@testing-library/react";
import React from "react";
import HeaderColumn from "../TulevaHeaderColumn/TulevaHeaderColumn";
import TulevaPage from "../TulevaPage/TulevaPage";
import { IBaseEntity } from "../../types";
import TulevaEditor from "../TulevaEditor/";
import TulevaMetaData from "../TulevaMetaData/TulevaMetaData";
import TulevaPageHeader from "../TulevaPageHeader/TulevaPageHeader";

const item2: IBaseEntity = {
  id: "test2",
  createdBy: { userID: "test2", name: "test2", email: "test2" },
  editedBy: { userID: "test2", name: "test2", email: "test2" },
  isDirty: false,
};

let items = [
  {
    id: "1",
    subject: "Wert 1",
    description:
      "A - Lorem ipsum dolor, sit amet consectetur adipisicing elit. Doloremque, recusandae neque! Corporis distinctio atque natus ea molestias in doloribus, ratione eius, tenetur aliquam saepe aspernatur dicta rem voluptas error nam.",
  },
  {
    id: "2",
    subject: "Wert 2",
    description:
      "B - Lorem ipsum dolor, sit amet consectetur adipisicing elit. Doloremque, recusandae neque! Corporis distinctio atque natus ea molestias in doloribus, ratione eius, tenetur aliquam saepe aspernatur dicta rem voluptas error nam.",
  },
  {
    id: "3",
    subject: "Wert 3",
    description:
      "E - Lorem ipsum dolor, sit amet consectetur adipisicing elit. Doloremque, recusandae neque! Corporis distinctio atque natus ea molestias in doloribus, ratione eius, tenetur aliquam saepe aspernatur dicta rem voluptas error nam.",
  },
];

describe("header column matches snapshot", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <HeaderColumn
        sortField="test"
        sortDescending={true}
        sortState="test"
        onClick={() => {}}
      >
        Test
      </HeaderColumn>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("header page matches snapshot", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<TulevaMetaData item={item2} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Header Column", () => {
  it("is truthy", () => {
    expect(HeaderColumn).toBeTruthy();
  });
});

describe("Page Header", () => {
  it("is truthy", () => {
    expect(TulevaPageHeader).toBeTruthy();
  });
});

describe("Meta Data", () => {
  it("is truthy", () => {
    expect(TulevaMetaData).toBeTruthy();
  });
});

describe("Page", () => {
  it("is truthy", () => {
    expect(TulevaPage).toBeTruthy();
  });
});

describe("Tuleva Editor", () => {
  it("is truthy", () => {
    expect(TulevaEditor).toBeTruthy();
  });
});

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const renderViewer = (item: any) => {
  return <></>;
};

const renderEditor = (item: any) => {
  return <>{item}</>;
};

const renderListHeader = () => {
  return <></>;
};

const renderFilter = () => {
  return <></>;
};
describe("header page matches snapshot", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <TulevaPage
        dataSource={items}
        loading={false}
        onAddNew={() => {}}
        onDelete={() => {}}
        onExit={() => {}}
        onSave={() => {}}
        onRenderView={(item) => renderViewer(item)}
        onRenderEdit={(item) => renderEditor(item)}
        onRenderListHeader={() => renderListHeader()}
        onRenderFilter={() => renderFilter()}
        createRights={true}
        deleteRights={false}
        editRights={false}
        listRights={true}
        caption={"Tests"}
        helpTopic="news"
        showAsModal={true}
        hideCopyButton={true}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
