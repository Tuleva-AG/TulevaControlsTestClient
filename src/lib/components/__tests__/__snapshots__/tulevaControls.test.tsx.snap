// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header column matches snapshot matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Test
    <span
      aria-label="caret-down"
      class="anticon anticon-caret-down"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="caret-down"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 1024 1024"
        width="1em"
      >
        <path
          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`header page matches snapshot matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <div>
          Erstellt am: Jan 3, 2023 1:55 PM von: test2
        </div>
      </div>
      <div>
        <div>
          Geändert am: Jan 3, 2023 1:55 PM von: test2
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`header page matches snapshot matches snapshot 2`] = `
<DocumentFragment>
  <div>
    <div
      class="undefined undefined"
    >
      <div>
        <div>
          <h2>
            Tests
          </h2>
        </div>
      </div>
      <div>
        <div />
      </div>
      <div>
        <div />
        <div
          class="undefined undefined"
        >
          <button
            class="ant-btn ant-btn-circle ant-btn-default ant-btn-icon-only"
            title="Neu"
            type="button"
          >
            <span
              aria-label="plus-circle"
              class="anticon anticon-plus-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="plus-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                />
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                />
              </svg>
            </span>
          </button>
        </div>
        <div />
      </div>
    </div>
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-list ant-list-sm ant-list-split ant-list-bordered"
        >
          <div
            class="ant-list-header"
          >
            <div>
              <div />
            </div>
            <div />
          </div>
          <div
            class="ant-spin-nested-loading"
          >
            <div
              class="ant-spin-container"
            >
              <ul
                class="ant-list-items"
              >
                <li
                  class="ant-list-item"
                >
                  <div>
                    <div>
                      <div
                        class="undefined undefined"
                      >
                        <div />
                      </div>
                    </div>
                    <div>
                      <div>
                        <span
                          class="ant-popover-disabled-compatible-wrapper"
                          style="display: inline-block; cursor: not-allowed;"
                        >
                          <button
                            class="ant-btn ant-btn-circle ant-btn-default ant-btn-icon-only"
                            disabled=""
                            style="pointer-events: none;"
                            title="Eintrag löschen..."
                            type="button"
                          >
                            <span
                              aria-label="delete"
                              class="anticon anticon-delete"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="delete"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </div>
                    </div>
                  </div>
                </li>
                <li
                  class="ant-list-item"
                >
                  <div>
                    <div>
                      <div
                        class="undefined undefined"
                      >
                        <div />
                      </div>
                    </div>
                    <div>
                      <div>
                        <span
                          class="ant-popover-disabled-compatible-wrapper"
                          style="display: inline-block; cursor: not-allowed;"
                        >
                          <button
                            class="ant-btn ant-btn-circle ant-btn-default ant-btn-icon-only"
                            disabled=""
                            style="pointer-events: none;"
                            title="Eintrag löschen..."
                            type="button"
                          >
                            <span
                              aria-label="delete"
                              class="anticon anticon-delete"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="delete"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </div>
                    </div>
                  </div>
                </li>
                <li
                  class="ant-list-item"
                >
                  <div>
                    <div>
                      <div
                        class="undefined undefined"
                      >
                        <div />
                      </div>
                    </div>
                    <div>
                      <div>
                        <span
                          class="ant-popover-disabled-compatible-wrapper"
                          style="display: inline-block; cursor: not-allowed;"
                        >
                          <button
                            class="ant-btn ant-btn-circle ant-btn-default ant-btn-icon-only"
                            disabled=""
                            style="pointer-events: none;"
                            title="Eintrag löschen..."
                            type="button"
                          >
                            <span
                              aria-label="delete"
                              class="anticon anticon-delete"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="delete"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </div>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
